sdks/go/test/integration/xlang/expansion_test.go

```diff
@@ -33,16 +34,21 @@ const (
 	expansionPort = "8097"
 )
 
-func checkPort(t *testing.T, port string) {
-	ping := exec.Command("nc", "-vz", "localhost", port)
-	output, err := ping.CombinedOutput()
-	if err != nil {
-		t.Errorf("failed to run ping to localhost:%v", port)
-	}
-	outputStr := string(output)
-	if strings.Contains(outputStr, "failed") {
-		t.Errorf("failed to connect to localhost:%v, got err %v", port, outputStr)
+func checkPort(t *testing.T, port string, durationSeconds float64) {
```

Comment:
Nit: use a time.Duration instead, then everything can work in the same units.

eg pass in a 15 * time.Second
&then precalculate interval and use it in the loop...

https://go.dev/play/p/hkmgtbCX191

Reply:
Good catch, should be fixed